src/core/a-entity.js

```diff
@@ -262,18 +267,17 @@ var proto = Object.create(ANode.prototype, {
       this.initComponentDependencies(name);
 
       if (isDependency && !isComponentDefined) {
-        // Add component if it is a dependency and not yet defined.
-        this.setAttribute(name, '');
-      } else {
-        if (this.isScene && !this.hasAttribute(name) && name in this.defaultComponents) {
-          // For scene default components, expose them in the DOM.
-          HTMLElement.prototype.setAttribute.call(this, name, this.defaultComponents[name]);
-        }
-        this.components[name] = new components[name].Component(this);
-        if (!this.paused) { this.components[name].play(); }
+        // Expose dependency component in the DOM.
+        HTMLElement.prototype.setAttribute.call(this, name, '');
+      }
```

Comment:
Did a bunch of debugging, the indirect use setAttribute to set up dependency components doesn't work. https://github.com/aframevr/aframe/pull/961#issuecomment-190437392